relay-general/src/store/transactions/processor.rs

```diff
@@ -84,6 +85,94 @@ impl<'r> TransactionsProcessor<'r> {
+        // doesn't exist already, scrub it, and remove it if we did nothing.
+        let previously_scrubbed =
+            matches!(span.data.value(), Some(data) if data.get("description.scrubbed").is_some());
+        if !previously_scrubbed {
+            dbg!(&span.description.clone().value());
+            if let Some(description) = span.description.clone().value() {
+                dbg!(&description);
+                span.data
+                    .value_mut()
+                    .get_or_insert_with(BTreeMap::new)
+                    .insert(
+                        "description.scrubbed".to_owned(),
+                        Annotated::new(Value::String(description.to_owned())),
+                    );
+            }
+        }
+
+        let mut scrubbed = false;
+
+        if let Some(data) = span.data.value_mut() {
+            if let Some(description) = data.get_mut("description.scrubbed") {
+                description.apply(|name, meta| {
+                    if let Value::String(s) = name {
+                        let result = self.name_config.rules.iter().find_map(|rule| {
+                            apply_tx_rule_to_string(Cow::Borrowed(s), rule.clone())
+                                .map(|new_name| (rule.pattern.compiled().pattern(), new_name))
+                        });
+
+                        if let Some((applied_rule, new_name)) = result {
+                            scrubbed = true;
+                            if *s != new_name {
+                                meta.add_remark(Remark::new(
+                                    RemarkType::Substituted,
+                                    // Setting a different format to not get
+                                    // confused by the actual `span.description`.
+                                    format!("description.scrubbed:{}", applied_rule),
+                                ));
+                                *name = Value::String(new_name);
+                            }
+                        }
+                    }
+
+                    Ok(())
+                })?;
+            }
+        }
+
+        if !previously_scrubbed && !scrubbed {
+            span.data
+                .value_mut()
+                .as_mut()
+                .and_then(|data| data.remove("description.scrubbed"));
+        }
+
+        Ok(())
+    }
+}
+
+/// Applies the transaction rule to the string, no matter what the string format is.
+///
+/// Transaction rules are generated on the assumption that they are only
+/// applied to transactions, which have a specific format. Span
+/// descriptions, however, have a different format, including the start of
+/// the string. As a workaround, we add a leading `**` to omit that first
+/// part of the string.
+fn apply_tx_rule_to_string(string: Cow<String>, mut rule: TransactionNameRule) -> Option<String> {
+    rule.pattern = LazyGlob::new(format!("**{}", rule.pattern.raw));
```

Comment:
This compiles the rule for every span, correct? Can we instead put another field `span_rules` on `name_config` or `self` which contains a separate set of `LazyGlob`s for spans?

Reply:
Alternative: Don't use different rules, just do a `split(" ")` on the description, or even check for case insensitive `(get|put|post|...) ` explicitly. Similar to how we add-and-remove a trailing slash before scrubbing, we can normalize the span description before applying the rule.

Reply:
I opted for the first version, since the second one would require more logic to also get rid of the subdomain properly.